### PR TITLE
Improve docs tooling

### PR DESCRIPTION
**Pull request by romanowski**
### Improve docs checking

- Fail if the provided path does not exists
- do not generate examples
- clean up checker and named it `sclicheck`
- run scripts as scripts not within scala
- add clear
- commands are not checked

### CODE REVIEW

File: .github/scripts/check_docs.sh [hunk 1]
```diff
@@ -8,5 +10,12 @@ export PATH=$dest:$PATH
...
+  then
+    toCheck="website/docs/cookbooks website/docs/commands"
+  else
+    toCheck=$@
+fi
+
 # adding --resources is a hack to get file watching for free on .md files
-scala-cli docs_checker/check.scala --resources docs/cookbooks $@ -- --dest examples docs/cookbooks 
\ No newline at end of file
+scala-cli sclicheck/sclicheck.scala --resources docs -- $toCheck
```
Comment (tpasternak) [inline .github/scripts/check_docs.sh:21]: Would be nice to make these paths absolute

File: sclicheck/Readme.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+Sclicheck `[sklicheck]` is a simple command line tool to verify documentation. 
+
+It uses regexes under the hood so in some cases we may not parse the file properly.
+
+Sclicheck extracts commands from `.md` file and then run this commands as defined within a file using a single workspace. Commands are not run in isolation by design. The whole point of the tool is to maintain a state of the current example and modify / test it using commands.
+
+Currently following commands are supported:
+ - [Write code](#write-code) - stores snippet into a file
+ - [Run code](#run-code) - runs provided commands as a bash scrip
```
Comment (tpasternak) [inline sclicheck/Readme.md]:
```suggestion
 - [Run code](#run-code) - runs provided commands as a bash script
```

File: sclicheck/sclicheck.scala [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  def name = toString.takeWhile(_ != '(')
+  def log = this match {
+    case _: Clear => ""
+    case Check(patterns, regex, _) =>
+      val kind = if regex then "regexes" else "patterns"
+      s"last output matches $kind: ${patterns.map(p => s"'$p'").mkString(", ")}"
+    case Run(cmd, shouldFail, _) =>
+      val prefix = if shouldFail then "[failure expected] " else ""
+      cmd.mkString(prefix, " ", "")
+    case Snippet(name, _, _) =>
```
Comment (tpasternak) [inline sclicheck/sclicheck.scala]: I'd rename it to `Write` to align with the doc of sclicheck

File: sclicheck/sclicheck.scala [hunk 4]
```diff
@@ -0,0 +1,292 @@
+// using scala 3.0.2
+// using lib com.lihaoyi::os-lib:0.7.8
+// using lib "com.lihaoyi::fansi:0.2.14"
+
+import scala.util.matching.Regex
+import scala.io.StdIn.readLine
+import fansi.Color.{Red, Blue, Green}
+
+val SnippetBlock  = """ *```[^ ]+ title=([\w\.\-\/_]+) *""".r
```
Comment (tpasternak) [inline sclicheck/sclicheck.scala]: Can title contain numbers?

File: sclicheck/sclicheck.scala [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+val CodeBlockEnds = """ *``` *""".r
+val BashCommand   = """ *```bash *(fail)? *""".r
+val CheckBlock    = """ *\<\!\-\- Expected(-regex)?: *""".r
```
Comment (tpasternak) [inline sclicheck/sclicheck.scala]: Do we need escaping of `-` here?

File: sclicheck/sclicheck.scala [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+  def parseArgs(args: Seq[String], options: Options): Options = args match
+    case Nil => options
+    case "--step" :: rest =>
+      parseArgs(rest, options.copy(step = true))
+    case "--stopAtFailure" :: rest =>
+      parseArgs(rest, options.copy(stopAtFailure = true))
+    case "--dest" :: dest :: rest =>
+      parseArgs(rest, options.copy(dest = Some(os.pwd / dest)))
+    case "--dest" :: Nil =>
```
Comment (tpasternak) [inline sclicheck/sclicheck.scala:290]: Will catch error when running `sclicheck --step --dest` but won't for `--dest --step`

**Review by tpasternak — APPROVED** [review]

File: sclicheck/sclicheck.scala (hunk 4 above)
Comment (romanowski) [reply sclicheck/sclicheck.scala]: good point!

**Pull request merged**